python/benchmark/benchmark/bench_nearest_neighbors.py

```diff
@@ -65,22 +113,27 @@ def run_once(
         if not is_single_col:
             input_cols = [c for c in train_df.schema.names]
 
+        query_df = train_df.sample(
+            withReplacement=False, fraction=fraction_sampled_queries, seed=seed
+        )
+
+        def cache_df(dfA: DataFrame, dfB: DataFrame) -> Tuple[DataFrame, DataFrame]:
+            dfA = dfA.cache()
+            dfB = dfB.cache()
+            dfA.count()
```

Comment:
Did you verify that count actually caches the dataframe?   I think sometimes it can be short-circuited via metadata (e.g. parquet files).

Reply:
Revised.